plugins/node/opentelemetry-instrumentation-pg/src/utils.ts

```diff
@@ -258,6 +264,50 @@ export function patchCallback(
   };
 }
 
+export function getPoolName(pool: PgPoolOptionsParams): string {
+  let poolName = '';
+  poolName += pool?.host ? `${pool.host}:` : 'unknown_host';
+  poolName += pool?.port ? `${pool.port}/` : 'unknown_port';
+  poolName += pool?.database ? `${pool.database}` : 'unknown_database';
+
+  return poolName.trim();
+}
+
+export interface poolConnectionsCounter {
+  used: number;
+  idle: number;
+  pending: number;
+}
+
+export function updateCounter(
+  pool: PgPoolExtended,
+  connectionCount: UpDownCounter,
+  connectionPendingRequests: UpDownCounter,
+  latestCounter: poolConnectionsCounter
+): poolConnectionsCounter {
+  const poolName = getPoolName(pool.options);
+  const all = pool.totalCount;
+  const pending = pool.waitingCount;
+  const idle = pool.idleCount;
+  const used = all - idle;
+
+  connectionCount.add(used - latestCounter.used, {
+    [SEMATTRS_CLIENT_CONNECTION_STATE]: 'used',
+    [SEMATTRS_CLIENT_CONNECTION_POOL_NAME]: poolName,
+  });
+
+  connectionCount.add(idle - latestCounter.idle, {
+    [SEMATTRS_CLIENT_CONNECTION_STATE]: 'idle',
+    [SEMATTRS_CLIENT_CONNECTION_POOL_NAME]: poolName,
+  });
+
+  connectionPendingRequests.add(pending - latestCounter.pending, {
+    [SEMATTRS_CLIENT_CONNECTION_POOL_NAME]: poolName,
+  });
+
+  return { used: used, idle: idle, pending: pending };
+}
+
```

Comment:
I can always bring this feedback to the group. For this PR I'll keep as the `UpDownCounter` and we can always change if comes to it (since it's experimental)